drivers/bluetooth/hci/userchan.c

```diff
@@ -185,42 +196,72 @@ static int uc_send(struct net_buf *buf)
 	return 0;
 }
 
-static int user_chan_open(uint16_t index)
+static int user_chan_open(void)
 {
-	struct sockaddr_hci addr;
 	int fd;
 
-	fd = socket(PF_BLUETOOTH, SOCK_RAW | SOCK_CLOEXEC | SOCK_NONBLOCK,
-		    BTPROTO_HCI);
-	if (fd < 0) {
-		return -errno;
-	}
+	if (hci_socket) {
+		struct sockaddr_hci addr;
+
+		fd = socket(PF_BLUETOOTH, SOCK_RAW | SOCK_CLOEXEC | SOCK_NONBLOCK,
+				BTPROTO_HCI);
+		if (fd < 0) {
+			return -errno;
+		}
+
+		(void)memset(&addr, 0, sizeof(addr));
+		addr.hci_family = AF_BLUETOOTH;
+		addr.hci_dev = bt_dev_index;
+		addr.hci_channel = HCI_CHANNEL_USER;
+
+		if (bind(fd, (struct sockaddr *)&addr, sizeof(addr)) < 0) {
+			int err = -errno;
+
+			close(fd);
+			return err;
+		}
+	} else {
+		struct sockaddr_in addr;
+
+		fd = socket(AF_INET, SOCK_STREAM, 0);
+		if (fd < 0) {
+			return -errno;
+		}
 
-	(void)memset(&addr, 0, sizeof(addr));
-	addr.hci_family = AF_BLUETOOTH;
-	addr.hci_dev = index;
-	addr.hci_channel = HCI_CHANNEL_USER;
+		addr.sin_family = AF_INET;
+		addr.sin_port = htons(port);
+		if (inet_pton(AF_INET, ip_addr, &(addr.sin_addr)) <= 0) {
+			close(fd);
+			return -errno;
+		}
 
-	if (bind(fd, (struct sockaddr *)&addr, sizeof(addr)) < 0) {
-		int err = -errno;
+		if (connect(fd, (struct sockaddr *)&addr, sizeof(addr)) < 0) {
+			int err = -errno;
 
-		close(fd);
-		return err;
+			close(fd);
+			return err;
+		}
 	}
 
+
 	return fd;
 }
 
 static int uc_open(void)
 {
-	if (bt_dev_index < 0) {
+	if (!arg_found) {
```

Comment:
This was already here, but anyhow: This if will never be true, as btuserchan_check_arg() would have errored out before otherwise.